share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
I also think that Option 1 is risky, even if it should happen to work :-)

But we still have a bit of investigation to do in order to determine whether we could make Option 2 portable. So far we have identified a small set of potential guard macros for which this might work on whatever version of Linux you are using, and my version of OS X (Yosemite). But is there any way of knowing whether those same guard macros would also work with older (or newer) versions of Linux and OS X? How about other systems, like Cygwin, Free BSD, etc. If we could at least determine that those headers are fairly stable between system versions, then perhaps maintenance would be feasible. Otherwise things could get tedious.

My key concern at this point is to avoid baking in platform dependencies to SMACK.